authors/views.py

```diff
@@ -0,0 +1,65 @@
+from django.shortcuts import render
+from django.http import HttpResponse, JsonResponse
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.parsers import JSONParser
+from rest_framework.generics import GenericAPIView
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from authors.models import Author 
+from authors.serializers import AuthorSerializer
+
+class AuthorView(GenericAPIView):
+    pagination_class = PageNumberPagination
+    serializer_class = AuthorSerializer
+    queryset = Author.objects.all()
+
+    def get(self, request):
+
+        authors = Author.objects.all()
+        serializer = AuthorSerializer(authors, many=True)
+        return Response(serializer.data)
+    # POST request
+    # Fields from client payload are: displayName, github, profileImage
+    # Fields filled in by server: host, url, id
+    def post(self, request):
+        data = JSONParser().parse(request)
+        data = fillAuthorJSONPayload(request, data)
```

Comment:
We do need this function so we can fill in the host and url of the Author model

Reply:
I dont think we should do this, im not sure because we might want to leave this option open so that we can add any host we want rather than making the host based on the request, if that makes sense.

So for example, lets say the front end detects oh we have a remote author, lets add it to our database, they send a post request to
http://localhost:8000/authors/
they should be able to say host="prestonsnode.com"

maybe im completely crazy, let me know if thats the case